omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/PatientAllergyResource2_0.java

```diff
@@ -0,0 +1,238 @@
+		if (parentUniqueId != null && uuid != null) {
+			// DELETE /patient/UUID/allergy/ALLERGYUUID means to delete one allergy from patient
+			Allergy allergy = getByUniqueId(uuid);
+			Context.getPatientService().voidAllergy(allergy, reason);
+		} else if (parentUniqueId != null && uuid == null) {
+			// DELETE /patient/UUID/allergy/ means to delete all allergies from patient, back to Unknown Allergies state
+			Patient patient = Context.getPatientService().getPatientByUuid(parentUniqueId);
+			Allergies allergies = Context.getPatientService().getAllergies(patient);
+			for (Allergy allergy : allergies) {
+				Context.getPatientService().voidAllergy(allergy, reason);
+			}
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#purge(java.lang.Object,
+	 *      org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void purge(Allergy newAllergy, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#purge(java.lang.Object,
+	 *      java.lang.String, org.openmrs.module.webservices.rest.web.RequestContext)
+	 */
+	@Override
+	public void purge(String parentUniqueId, String uuid, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceHandler#save(java.lang.Object)
+	 */
+	@Override
+	public Allergy save(Allergy newAllergy) {
+		Context.getPatientService().saveAllergy(newAllergy);
+		return newAllergy;
+	}
+
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource#newDelegate()
+	 */
+	@Override
+	public Allergy newDelegate() {
+		return new Allergy();
+	}
+
+	/**
+	 * Gets the display string for a person name.
+	 *
+	 * @param Allergy the allergy object.
+	 * @return the display string.
+	 */
+	@PropertyGetter("display")
+	public String getDisplayString(Allergy allergy) {
+		if (allergy.getAllergen().isCoded()) {
+			return allergy.getAllergen().getCodedAllergen().getDisplayString();
+		} else {
+			return allergy.getAllergen().getNonCodedAllergen();
+		}
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingSubResourcep[ut(java.lang.String, SimpleObject, RequestContext)
+	 * PUT /patient/{patient_uuid}/allergy sets allergyStatus of patient to Allergies.NO_KNOWN_ALLERGIES
+	 */
+	@Override
+	public void put(String parentUniqueId, SimpleObject post, RequestContext context) throws ResponseException {
+		Patient patient = Context.getPatientService().getPatientByUuid(parentUniqueId);
+		if (patient == null) {
+			throw new ObjectNotFoundException();
+		}
+		
+		if (post.size() != 0)  {
+			// not allowed to actually post any data
+			throw new ObjectNotFoundException();

```

Comment:
@dkayiwa see https://talk.openmrs.org/t/questions-about-resources-in-allergy-ui-module/3750/26?u=jdegraft, does ObjectNotFoundException not return a 400 error or what should the right exception be?